packages/types/test/unit/union.test.ts

```diff
@@ -0,0 +1,15 @@
+import {ssz} from "../../src";
+
+describe("union", () => {
+  it("struct hashTreeRoot", () => {
+    const defaultValue = ssz.merge.ExecutionPayload.defaultValue();
+    defaultValue.transactions.push(Buffer.alloc(512, 0));
```

Comment:
for Union, value is actually a wrapped value, it needs a selector (uint) specifying the type used for that Union. For example this is from the spec test (which was passed)

```
 transactions:
 - {selector: 0, value: '0xfa61e831ac3cab767c448be074e4f555537ea08bf0341d8aba3682d6b78313d9442e2ef03905e61f65433fac    13d591b5a84af52249d7dc78dfe14276820a2ed6bd164e8f6e49abb32a05ce809d772a6233977db550da180fc4f74d54d7847895568963db44    1a6745ffa3197daef756976993185af5bafc7518f34799cac1d2a1ca97fc7fd2effd36841ac418ef8826b8a5494961dc0b068742f2d04e873c    f512bcb02c748eafc1e6'}
```

this works
```
// @ts-ignore
    defaultValue.transactions.push({selector: 0, value: Buffer.alloc(512, 0)});
```